src/main/webapp/js/common/react/components/AuthButton.js

```diff
@@ -0,0 +1,83 @@
+import checkTesting from '../../../checkTesting.js';
+import standardizeFetchErrors from '../../../fetch_handler.js';
+checkTesting();
+
+/**
+ * Login/Logout button that fetches user authorization data to automatically
+ * update its appearance.
+ * @extends React.Component
+ */
+export default class AuthButton extends React.Component {
+  /**
+   * Create the AuthButton react component.
+   * @param {{}} props 
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isFetching: true,
+    };
+  }
+
+  /**
+   * Loads user authorization data after React has initialized the component.
+   * @override
+   */
+  componentDidMount() {
```

Comment:
Q: Does this function only call another function? (if so, could you just move the body of 'fetchAuthorizationStatus' here? or will you need that function other places)

Reply:
Good point. I think this is a remnant of thinking I was going to need it more than once. Will update!